pkg/controller/common/helpers.go

```diff
@@ -1,25 +1,159 @@
 package common
 
 import (
+	"bytes"
+	"encoding/json"
+	"fmt"
 	"io/ioutil"
 	"reflect"
+	"sort"
 
+	ign "github.com/coreos/ignition/config/v2_2"
 	igntypes "github.com/coreos/ignition/config/v2_2/types"
 	validate "github.com/coreos/ignition/config/validate"
 	"github.com/golang/glog"
 	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
 	errors "github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
 )
 
-// NewIgnConfig returns an empty ignition config with version set as latest version
-func NewIgnConfig() igntypes.Config {
+const (
+	kernelTypeDefault  = "default"
+	kernelTypeRealtime = "realtime"
+)
+
+// DecodeIgnitionConfigSpecV2 decodes byte slices to Ignition Config Spec v2 or errors out
+func DecodeIgnitionConfigSpecV2(data []byte) (*igntypes.Config, error) {
```

Comment:
Actually, now I think about it, wouldn't we need some sort of function to first detect ignition version in a `data []byte` object? If we just have a `MachineConfig object` saved as `mc.Spec.Config.Raw` we wouldn't know what to decode it to.

Reply:
Yes we will need such a function! But since we only support V2 at this time, contents are always assumed to be V2 (or error out otherwise).

I thought it'd be overkill to add it already, but we'll definitely need it for dual-support. It could look something like this:
```
// GetSupportedIgnitionVersionFromRawConfig decodes byte slices to get the Ignition version contained or errors out
func GetSupportedIgnitionVersionFromRawConfig(data []byte) (string, error) {
	var cfgMap map[string]*json.RawMessage
	err := json.Unmarshal(data, &cfgMap)
	if err != nil {
		return "", err
	}
	type Ignition struct {
		Version string
	}
	var ign Ignition
	err = json.Unmarshal(*cfgMap["ignition"], &ign)
	if err != nil {
		return "", err
	}

	if ign.Version >= "2.0.0" && ign.Version <= igntypes.MaxVersion.String() {
		return "v2", nil
	}

	return "", fmt.Errorf("Unsupported Ignition Config Spec Version detected")
}
```

Reply:
> Yes we will need such a function! But since we only support V2 at this time, contents are always assumed to be V2 (or error out otherwise).

I think that's Kirsten WIP PR

Reply:
Agree with runcom. Scope of this PR should stay V2, we can merge then I can make whatever v3 changes are needed in my PR (also giving this PR some soak time).